libnavui-androidauto/src/main/java/com/mapbox/androidauto/screencoordinator/MapboxScreenManager.kt

```diff
@@ -0,0 +1,231 @@
+package com.mapbox.androidauto.screencoordinator
+
+import androidx.annotation.UiThread
+import androidx.car.app.CarContext
+import androidx.car.app.Screen
+import androidx.car.app.ScreenManager
+import androidx.car.app.Session
+import androidx.car.app.model.Template
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.androidauto.internal.logAndroidAuto
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
+import kotlin.jvm.Throws
+
+/**
+ * The Mapbox Navigation Android Auto SDK is prepared with a default experience. This object allows
+ * you to customize the experience to meet your needs.
+ */
+class MapboxScreenManager {
+    private var carContext: CarContext? = null
+    private var screenManager: ScreenManager? = null
+    private var lifecycleOwner: LifecycleOwner? = null
+    private val screenFactoryMap = mutableMapOf<String, MapboxScreenFactory>()
+    private var coroutineScope: CoroutineScope? = null
+
+    private val lifecycleObserver: DefaultLifecycleObserver = object : DefaultLifecycleObserver {
+        override fun onCreate(owner: LifecycleOwner) {
+            val carContext = requireCarContext()
+            screenManager = carContext.getCarService(ScreenManager::class.java)
+            coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+            coroutineScope?.launch {
```

Comment:
Can we use the built-in scope?
```suggestion
            owner.lifecycleScope.launch {
```

Reply:
oh yeah it's a parameter lol